airflow/providers/amazon/aws/hooks/glue_session.py

```diff
@@ -0,0 +1,265 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import asyncio
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class GlueSessionHook(AwsBaseHook):
+    """
+    Interact with AWS Glue Session.
+
+    Provide thick wrapper around :external+boto3:py:class:`boto3.client("glue") <Glue.Client>`.
+
+    :param id: session id
+    :param desc: session description
+    :param iam_role_name: AWS IAM Role for Glue Session Execution. If set `iam_role_arn` must equal None.
+    :param iam_role_arn: AWS IAM Role ARN for Glue Session Execution, If set `iam_role_name` must equal None.
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Session
+    :param create_session_kwargs: Extra arguments for Glue Session Creation
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        - :class:`airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    def __init__(
+        self,
+        session_id: str | None = None,
+        desc: str | None = None,
+        iam_role_name: str | None = None,
+        iam_role_arn: str | None = None,
+        num_of_dpus: int | None = None,
+        create_session_kwargs: dict | None = None,
+        session_poll_interval: int | float = 6,
+        *args,
+        **kwargs,
+    ) -> None:
+        self.session_id = session_id
+        self.desc = desc
+        self.role_name = iam_role_name
+        self.role_arn = iam_role_arn
+        self.create_session_kwargs = create_session_kwargs or {}
+        self.session_poll_interval = session_poll_interval
+
+        worker_type_exists = "WorkerType" in self.create_session_kwargs
+        num_workers_exists = "NumberOfWorkers" in self.create_session_kwargs
+
+        if self.role_arn and self.role_name:
+            raise ValueError("Cannot set iam_role_arn and iam_role_name simultaneously")
+        if worker_type_exists and num_workers_exists:
+            if num_of_dpus is not None:
+                raise ValueError("Cannot specify num_of_dpus with custom WorkerType")
+        elif not worker_type_exists and num_workers_exists:
+            raise ValueError("Need to specify custom WorkerType when specifying NumberOfWorkers")
+        elif worker_type_exists and not num_workers_exists:
+            raise ValueError("Need to specify NumberOfWorkers when specifying custom WorkerType")
+        elif num_of_dpus is None:
+            self.num_of_dpus: int | float = 10
+        else:
+            self.num_of_dpus = num_of_dpus
+
+        kwargs["client_type"] = "glue"
```

Comment:
This looks like forth hook around [`glue` client](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue.html):
- [airflow.providers.amazon.aws.hooks.glue.GlueJobHook](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_modules/airflow/providers/amazon/aws/hooks/glue.html#GlueJobHook)
- [airflow.providers.amazon.aws.hooks.glue_catalog.GlueCatalogHook](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_modules/airflow/providers/amazon/aws/hooks/glue_catalog.html#GlueCatalogHook)
- [airflow.providers.amazon.aws.hooks.glue_crawler.GlueCrawlerHook](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/glue_crawler/index.html#airflow.providers.amazon.aws.hooks.glue_crawler.GlueCrawlerHook)

Reply:
We have set a strange precedent here, if they're using the same client then the same hook class could cover them all. Or at least put them all in the same module?

But also we have the precedent now, so I'm not opposed to just following it again here.